app/models/spree/order_decorator.rb

```diff
@@ -40,4 +40,12 @@ def avatax_order_after_save
     end
   end
 
+  def line_items_with_tax_rates
```

Comment:
What you're doing here is _essentially_ a scope on `Spree::LineItem`. Could it be implemented as such? Along the lines of:

```
class Spree::LineItem
  def self.with_tax_rates
    lis = select do |item|
      li.product.tax_category.tax_rates.any?
    end

    where(id: lis.map(&:id))
   end
end
```

Then you could further extract the predicate test for a tax-rate to simplify the scope:

```
class Spree::LineItem
  def taxable?
    product.tax_category.tax_rates.any?
  end
end
```

This improves the implementation in a few ways:

- isolates the logic for taxability, and provides a more reusable and overridable interface;
- separates the concerns for scoping line-items and the taxability;
- the scope returns an ActiveRelation, which makes it chainable;

There's probably even a way to perform this test entirely with SQL, but that'd require some gnarly ARel wrangling, I think.